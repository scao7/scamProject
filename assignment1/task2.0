8076 -0.4200000000 1.4500000000
