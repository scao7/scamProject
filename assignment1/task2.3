2085 0.7500000000 0.6700000000
