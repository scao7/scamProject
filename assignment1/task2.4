3387 0.8000000000 1.0700000000
