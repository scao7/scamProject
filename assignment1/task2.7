2280 -0.0500000000 1.2700000000
