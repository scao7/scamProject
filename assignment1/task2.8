3582 0.000000e+00 -0.3300000000
