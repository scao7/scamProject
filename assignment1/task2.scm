(define (mandelbrot threshold)
  ; takes point (x, y) and checks if it
  ; is within the mandlebrot set with threshold
  ; of certainty
  (lambda (x y)
    ; recursive function to check divergence
    (define (iter r s t)
      ; check for threshold reached
      (if (= t threshold) 
        0
        ; check for divergence
        (if (< 2 (sqrt (+ (* r r) (* s s))))
          ; if diverges, return t (the number of iterations checked)
          t
          ; otherwise, iterate one level deeper
          (iter 
            ; r value
            (+ x (- (* r r) (* s s)))
            ; s value
            (+ y (* r s 2))
            ; t value
            (+ 1 t)
            )
          )
        )
      )
    ; call with default values 0.0 for r and 0.0 for s
    (iter 0.0 0.0 0)
    )
  )

; (0, 0) should yeild 0
(define (main)
  (setPort (open (getElement ScamArgs 1) 'read))
  (define arg (readExpr))
  (define arg1 (readExpr))
  (define arg2 (readExpr))
  (println "((mandelbrot " arg ") " arg1 arg2 ")) is " ((mandelbrot arg) arg1 arg2)) 
  ;(inspect ((mandelbrot arg) arg1 arg2))
  )