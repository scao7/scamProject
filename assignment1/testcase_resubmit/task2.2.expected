((mandelbrot 777) 0.6900000000 1.2600000000) is 2
